rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -261,7 +264,8 @@ rmw_destroy_client(rmw_node_t * node, rmw_client_t * client)
     }
     delete info;
   }
-  delete (client);
+  rmw_free(const_cast<char *>(client->service_name));
```

Comment:
> if you think of rmw_free() as a "black box" function that you don't know anything about, then it is helpful to be conservative and not pass NULL into there.

+1 these functions are expected to be overriden by users wanting different memory strategies using alternative allocators. So I think it makes sense to sanity check it before